client/CPlayerInterface.cpp

```diff
@@ -182,32 +189,14 @@ void CPlayerInterface::playerEndsTurn(PlayerColor player)
 void CPlayerInterface::playerStartsTurn(PlayerColor player)
 {
 	EVENT_HANDLER_CALLED_BY_CLIENT;
-
-	movementController->onPlayerTurnStarted();
-
-	if(GH.windows().findWindows<AdventureMapInterface>().empty())
-	{
-		// after map load - remove all active windows and replace them with adventure map
-		GH.windows().clear();
-		GH.windows().pushWindow(adventureInt);
-	}
-
-	// close window from another player
-	if(auto w = GH.windows().topWindow<CInfoWindow>())
-		if(w->ID == QueryID::NONE && player != playerID)
-			w->close();
-	
-	// remove all dialogs that do not expect query answer
-	while (!GH.windows().topWindow<AdventureMapInterface>() && !GH.windows().topWindow<CInfoWindow>())
-		GH.windows().popWindows(1);
-

```

Comment:
can you please confirm, that different scenarios with turn timer expiring (including messages, hot-seat turn passing, etc) work as expected and don't trigger assertions after your change?

Reply:
Will recheck, although this block was not removed but moved to other method(s)